.github/workflows/build-test-osx-m1.yaml

```diff
@@ -52,35 +71,60 @@ jobs:
           name: semgrep-m1-${{ github.sha }}
 
   build-wheels-m1:
-    runs-on: [self-hosted, macOS, ARM64, aws]
+    runs-on:
+      [
+        "self-hosted",
+        "macOS",
+        "ARM64",
+        "ghcr.io/cirruslabs/macos-monterey-base:latest",
+      ]
     needs: [build-core-m1]
     steps:
+      - name: Setup runner directory
+        run: |
+          sudo mkdir /Users/runner
+          sudo chmod 750 /Users/runner
+          sudo chown -R admin:staff /Users/runner
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
       - uses: actions/checkout@v3
         with:
           submodules: true
       - uses: actions/download-artifact@v3
         with:
           name: semgrep-m1-${{ github.sha }}
-      - run: |
+      - name: Build Wheels
+        run: |
           unzip artifacts.zip
           cp artifacts/semgrep-core cli/src/semgrep/bin
-          ./scripts/build-wheels.sh
+          ./scripts/build-wheels.sh --plat-name macosx_11_0_arm64
```

Comment:
again a link explaining what is this magic macosx_ string would be useful.
Is it something related to Cirrus? To Setuptools called in build-wheels.sh?